File: 프로그래머스/0/120863. 다항식 더하기/다항식 더하기.js
function solution(polynomial) {
    const arr = polynomial.split(" + ");
    const x = [], n = [];

    arr.forEach((v) => v.includes("x") ? x.push(v) : n.push(v));

    const nOp = x.filter((v) => v.match(/\dx/g)).map((v) => v.substr(0, v.length - 1));
    const sum1 = nOp.reduce((a, c) => a + Number(c), 0);

    const sum2 = x.length - nOp.length;

    const sum3 = n.reduce((a, c) => a + Number(c), 0);

    if (polynomial === "x") {
        return "x";
    } else if (sum3 === 0) {
        return (sum1 + sum2) + "x"
    } else {
        if (x.length === 0) {
            return sum3 + "";
        }
        else if ((sum1 + sum2) === 1) {
            return "x" + " + " + sum3;
        }
        else {
            return (sum1 + sum2) + "x" + " + " + sum3;
        }
    }
}

File: 프로그래머스/0/120863. 다항식 더하기/다항식 더하기.py
def solution(polynomial):
    x_coeff_sum = 0
    constant_sum = 0

    terms = polynomial.split(" + ")

    for term in terms:
        if 'x' in term:
            if term == 'x':
                x_coeff_sum += 1
            else:
                x_coeff_sum += int(term[:-1])
        else:
            constant_sum += int(term)

    result_parts = []
    if x_coeff_sum > 0:
        if x_coeff_sum == 1:
            result_parts.append('x')
        else:
            result_parts.append(f'{x_coeff_sum}x')

    if constant_sum > 0:
        result_parts.append(str(constant_sum))

    if not result_parts:
        return "0"
    
    return " + ".join(result_parts)
